src/CadViewer.tsx

```diff
@@ -75,10 +92,23 @@ const CadViewerInner = (props: any) => {
 
   const handleCameraPresetSelect = useCallback(
     (preset: CameraPreset) => {
+      // Stop auto-rotate when a preset is selected
+      setAutoRotate(false)
+      setAutoRotateUserToggled(true)
+
       setCameraPreset(preset)
       closeMenu()
+      lastPresetSelectTime.current = Date.now()
+
       if (preset === "Custom") return
+
+      isAnimatingRef.current = true
       cameraControllerRef.current?.animateToPreset(preset)
+
+      // Reset the animation flag after the animation would be complete
+      setTimeout(() => {
+        isAnimatingRef.current = false
+      }, 600) // Match this with the animation duration in useCameraController
```

Comment:
this logic is complicated for a top-level file like CadViewer, can you isolate the hook to a separate file for managing this... these files tend to grow uncontrollably if you just add additional logic